edge-modules/PaidModuleSample/src/PurchaseInfoProvider.cs

```diff
@@ -0,0 +1,1236 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace PaidModuleSample
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.IO;
+    using System.Linq;
+    using System.Net;
+    using System.Net.Http;
+    using System.Net.Http.Headers;
+    using System.Net.Sockets;
+    using System.Runtime.InteropServices;
+    using System.Security.Cryptography.X509Certificates;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Newtonsoft.Json;
+
+    public class PurchaseInfoProvider
```

Comment:
calling GetPurchaseAsync is what the customer should need to know about, they don't need to know the code of the method because it will be in the SDK. Everything else is helper functions to get the token and install edgeHub trustbundle. Yes, I think we should tell them is a workaround or temporary sample and it will probably change when the API is added to SDK.